src/logic/tests/calculate.test.js

```diff
@@ -0,0 +1,51 @@
+import calculate from '../calculate';
+import operate from '../operate';
+
```

Comment:
- [ ] You added tests for only the calculate and operate files. Kindly note that you are required to add snapshot tests as well. And this should be done for all your components.

- [ ] You are also required to add tests that simulate user interactions using the react testing library. Kindly do that.